compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -794,6 +794,8 @@ public void visit(BLangPackage pkgNode) {
 
         pkgNode.constants = removeDuplicateConstants(pkgNode);
 
+        rewrite(pkgNode.typeDefinitions, env);
```

Comment:
Can you briefly explain how changing the order fixes the issue?

Reply:
In the current implementation, we desugar the value of global variables after adding the init function, but for configurables, we desugar before adding them to the init function. 

In the `desugarGlobalVariables` function, we perform the following actions: if there's a global variable defined as `record{int x = 10;}  k = {}`, we create a variable declaration `record{int x = 10;}  k` and add it to the global variable list. Additionally, we create another variable reference `k = {}` and add it to the initialization function. 
The values of global variables are desugared during the desugaring process of the initialization function.(So global variables are working properly as expected)

In the current implementation, the desugaring of configurable variables' values takes place when visiting the global variables, rather than during the desugaring of the init function, as previously mentioned.

Due to this, we need to desugar type definitions before desugaring configurables.

Reply:
Can't we change configurables to be similar to global variables? Not suggesting that we do it, but checking on the feasibility. Changing the order can have unexpected effects, and therefore, updating configurables may be a safer choice.

Reply:
Yeah, I will check that.